src/main/java/frc/robot/subsystems/Launcher.java

```diff
@@ -0,0 +1,217 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+ 
+package frc.robot.subsystems;
+ 
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.ControlType;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import com.revrobotics.CANPIDController;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.LauncherConstants;
+ 
+public class Launcher extends SubsystemBase {
+ 
+  private final CANSparkMax m_leftLauncher;
+  private final CANSparkMax m_rightLauncher;
+ 
+  private final CANEncoder m_leftEncoder;
+  private final CANEncoder m_rightEncoder;
+
+  private CANPIDController m_pidController;
+  
+  //variables for constants
+  private double pVal;
+  private double iVal;
+  private double dVal;
+  private double izVal;
+  private double ffVal;
+  private double minVal;
+  private double maxVal;
+ 
+  /**
+   * Creates a new Launcher.
+   */
+  public Launcher() {
+    // Instantiate the motors.
+    m_leftLauncher = new CANSparkMax(LauncherConstants.kLeftLauncherMotorPort, MotorType.kBrushless);
+    m_rightLauncher = new CANSparkMax(LauncherConstants.kRightLauncherMotorPort, MotorType.kBrushless);
+ 
+    // Instantiate the encoder.
+    m_leftEncoder = m_leftLauncher.getEncoder();
+    m_rightEncoder = m_rightLauncher.getEncoder();
+ 
+    // Initialize the the motors.
+    motorInit(m_leftLauncher, false);
+    motorInit(m_rightLauncher, true);
```

Comment:
The inversion of a motor should be stored in the constants to be consistent with the formatting of the other subsystems.